checkov/terraform/checks/resource/aws/AbsNACLUnrestrictedIngress.py

```diff
@@ -0,0 +1,58 @@
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+from checkov.common.util.type_forcers import force_list
+from checkov.common.util.type_forcers import force_int
+
+
+class AbsNACLUnrestrictedIngress(BaseResourceCheck):
+    def __init__(self, check_id, port):
+        name = "Ensure no NACL allow ingress from 0.0.0.0:0 to port %d" % port
+        supported_resources = ['aws_network_acl', 'aws_network_acl_rule']
+        categories = [CheckCategories.NETWORKING]
+        super().__init__(name=name, id=check_id, categories=categories, supported_resources=supported_resources)
+        self.port = port
+
+    def scan_resource_conf(self, conf):
+        """
+
+            Return PASS if:
+            - The NACL doesnt allows restricted ingress access to the port
+            - The resource is an aws_network_acl of type 'ingress' that does not violate the check.
+
+            Return FAIL if:
+            - The the NACL allows unrestricted access to the port
+
+            Return UNKNOWN if:
+            - the resource is an NACL of type 'egress', OR
+
+        :param conf: aws_network_acl configuration
+        :return: <CheckResult>
+        """
+
+        if conf.get("ingress"):
+            ingress = conf.get("ingress")
+            for rule in ingress:
+                if not self.check_rule(rule):
+                    return CheckResult.FAILED
+            return CheckResult.PASSED
+        # maybe its an network_acl_rule
+        if conf.get("network_acl_id"):
+            if not conf.get("egress")[0]:
+                if not self.check_rule(conf):
+                    return CheckResult.FAILED
+            return CheckResult.PASSED
+
+        return CheckResult.UNKNOWN
+
+    def check_rule(self, rule):
+        if rule.get('cidr_block'):
+            if rule.get('cidr_block') == ["0.0.0.0/0"]:
+                if rule.get('action') == ["allow"] or rule.get('rule_action') == ["allow"]:
+                    if rule.get('from_port')[0] >= self.port and rule.get('to_port')[0] <= self.port:
```

Comment:
```suggestion
                    if rule.get('from_port')[0] <= self.port and rule.get('to_port')[0] >= self.port:
```
I think that's how it should be, can you either change one of the test cases or add a new one, which will test a range like, 10 - 100, then it should fail of course 😄

Reply:
wont that be invalid terraform to start with or am i not getting it?

Reply:
i should put some ranges in though

Reply:
sorry, what I meant was to add a test case like that
```hcl
resource "aws_network_acl_rule" "pass" {
  network_acl_id = aws_network_acl.pass.id
  rule_number    = 200
  egress         = false
  protocol       = "tcp"
  rule_action    = "allow"
  cidr_block     = "10.0.0.0/32"
  from_port      = 3300
  to_port        = 3400
}
```
and this should fail for the RDP check

Reply:
yeah i understand now, i put the ranges in. good catch